x/blob/README.md

```diff
@@ -2,49 +2,192 @@
 
-## State
+### Params
 
-The blob module doesn't maintain it's own state.
+```proto
+// Params defines the parameters for the module.
+message Params {
+  option (gogoproto.goproto_stringer) = false;
+  uint32 gas_per_blob_byte = 1
+      [ (gogoproto.moretags) = "yaml:\"gas_per_blob_byte\"" ];
+  uint64 gov_max_square_size = 2
+      [ (gogoproto.moretags) = "yaml:\"gov_max_square_size\"" ];
+}
+```
 
-When a `MsgPayForBlob` is processed, it consumes gas based on the blob size.
+#### `GasPerBlobByte`
 
-## Messages
+`GasPerBlobByte` is the amount of gas that is consumed per byte of blob data
+when a `MsgPayForBlob` is processed. Currently, the Default value is 8. This
+value is set below that of normal transaction gas consumption, which is 10.
+
+#### `GovMaxSquareSize`
 
-- [`MsgPayForBlob`](https://github.com/celestiaorg/celestia-app/blob/8b9c4c9d13fe0ccb6ea936cc26dee3f52b6f6129/proto/blob/tx.proto#L39-L44) pays for the blob to be included in the block.
+`GovMaxSquareSize` is the maximum size of a data square that is considered valid
+by the validator set. This value is superseded by the `MaxSquareSize`, which is
+hardcoded and cannot change without hardforking the chain. See
+[ADR021](../../docs/architecture/adr-021-restricted-block-size.md) for more
+details.
 
-## PrepareProposal
+## Messages
 
-When a block producer is preparing a block, they must perform an extra step for `BlobTx`s so that end-users can find the blob shares relevant to their submitted `BlobTx`. In particular, block proposers wrap the `BlobTx` in the PayForBlobs namespace with the index of the first share of the blob in the data square. See [Blob share commitment rules](../../specs/src/specs/data_square_layout.md#blob-share-commitment-rules) for more details.
+- [`MsgPayForBlobs`](https://github.com/celestiaorg/celestia-app/blob/v1.0.0-rc2/proto/celestia/blob/v1/tx.proto#L16-L31)
+  pays for a set of blobs to be included in the block. Blob transactions that contain
+  this `sdk.Msg` are also referred to as "PFBs".
+  this `sdk.Msg` are better known as "PFBs".
+
+```proto
+message MsgPayForBlobs {
+  string signer = 1;
+  repeated bytes namespaces = 2;
+  repeated uint32 blob_sizes = 3;
+  repeated bytes share_commitments = 4;
+  repeated uint32 share_versions = 8;
+}
+```
 
-Since `BlobTx`s can contain multiple blobs, the `BlobTx` is wrapped with one share index per blob in the transaction. The index wrapped transaction is called an [IndexWrapper](https://github.com/celestiaorg/celestia-core/blob/2d2a65f59eabf1993804168414b86d758f30c383/proto/tendermint/types/types.proto#L192-L198) and this is the type that gets marshalled and written to the PayForBlobNamespace.
+`MsgPayForBlobs` pays for the inclusion of a blob in the block and consists of the
+following fields:
+
+- signer: bech32 encoded signer address
+- namespace: namespace is a byte slice of length 33 where the first byte is the
+  namespaceVersion and the subsequent 32 bytes are the namespaceId.
```

Comment:
[blocking]

```suggestion
- namespace: namespace is a byte slice of length 29 where the first byte is the
  namespaceVersion and the subsequent 28 bytes are the namespaceId.
```

Reply:
ahh shoot, I purposfully didn't update since I saw https://github.com/celestiaorg/celestia-app/pull/1951/files :facepalm: but forgot that this was copied pasted. thanks for catching